moto/ec2/models.py

```diff
@@ -6351,6 +6500,152 @@ def describe_transit_gateway_vpc_attachments(
 
         return transit_gateways_attachments
 
+    def delete_transit_gateway_vpc_attachment(self, transit_gateway_attachment_id=None):
+        self.transit_gateway_attachments[
+            transit_gateway_attachment_id
+        ].state = "deleted"
+        transit_gateway_attachment = self.transit_gateway_attachments.pop(
+            transit_gateway_attachment_id
+        )
+        return transit_gateway_attachment
+
+    def modify_transit_gateway_vpc_attachment(
+        self,
+        add_subnet_ids=None,
+        options=None,
+        remove_subnet_ids=None,
+        transit_gateway_attachment_id=None,
+    ):
+        if remove_subnet_ids:
+            self.transit_gateway_attachments[
+                transit_gateway_attachment_id
+            ].subnet_ids = [
+                id
+                for id in self.transit_gateway_attachments[
+                    transit_gateway_attachment_id
+                ].subnet_ids
+                if id not in remove_subnet_ids
+            ]
+        if options:
+            self.transit_gateway_attachments[
+                transit_gateway_attachment_id
+            ].options.update(options)
+
+        if add_subnet_ids:
+            self.transit_gateway_attachments[
+                transit_gateway_attachment_id
+            ].subnet_ids = add_subnet_ids
+
+        return self.transit_gateway_attachments[transit_gateway_attachment_id]
+
+    def set_attachment_association(
+        self, transit_gateway_attachment_id=None, transit_gateway_route_table_id=None
+    ):
+        self.transit_gateway_attachments[transit_gateway_attachment_id].association = {
+            "state": "associated",
+            "transitGatewayRouteTableId": transit_gateway_route_table_id,
+        }
+
+    def set_attachment_propagation(
+        self, transit_gateway_attachment_id=None, transit_gateway_route_table_id=None
+    ):
+        self.transit_gateway_attachments[transit_gateway_attachment_id].association = {
+            "state": "enabled",
+            "transitGatewayRouteTableId": transit_gateway_route_table_id,
+        }
+
+    def disable_attachment_propagation(self, transit_gateway_attachment_id=None):
+        self.transit_gateway_attachments[transit_gateway_attachment_id].association[
+            "state"
+        ] = "disabled"
+
+
+class TransitGatewayRelations(object):
+    # this class is for TransitGatewayAssociation and TransitGatewayPropagation
+    def __init__(
+        self,
+        backend,
+        transit_gateway_attachment_id=None,
+        transit_gateway_route_table_id=None,
+        state=None,
+    ):
+        self.transit_gateway_attachment_id = transit_gateway_attachment_id
+        self.transit_gateway_route_table_id = transit_gateway_route_table_id
+        self.resource_id = backend.transit_gateway_attachments[
```

Comment:
nit: similar comment as above - let's try to avoid duplicate code like this one in general.

better:
```
    tgw_attachment = backend.transit_gateway_attachments[transit_gateway_attachment_id]
    self.resource_id = tgw_attachment.resource_id
    self.resource_type = tgw_attachment.resource_type
```